File: Data Structures & Algorithms/permutations/submission-0.cpp
class Solution {
public:
    vector<vector<int>> permute(vector<int>& nums) {
        if(nums.empty()){
            return {{}};
        }

        vector<int> temp = vector<int>(nums.begin()+1,nums.end());

        vector<vector<int>> perm = permute(temp);
        vector<vector<int>> res;

        for(const auto& p  : perm){
            for(int i=0;i<=p.size();i++){
                vector<int> copy = p ;
                copy.insert(copy.begin()+i,nums[0]);
                res.push_back(copy);
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [] 
        for i in range(len(nums)):
            k=1
            for j in range(len(nums)):
                if i!=j:
                    k*=nums[j]
                
            res.append(k)

        return res


        